src/compiler/diagnosticMessages.json

```diff
@@ -3543,6 +3547,11 @@
         "code": 6196,
         "reportsUnnecessary": true
     },
+    "Resolve module name imported with '.json' extension to the json source file.": {

```

Comment:
I think this message could be improved -- not obvious what the alternative is. `"Type-check imported '.json' files"` might be a better explanation. The same for the error message -- it's not the module resolution that users will care about, but the fact that we typecheck the contents of the file.

Reply:
But that's not what it means though.. We wont even look for .json files if the flag is on. So its not just typechecking but picking up the modules?

Reply:
How about "Include '.json' files in the program"?
The reason is that module resolution seems like more of an internal issue that most users won't care about. But they will care if their '.json' files are getting compiled or not.

Reply:
Assuming that `.json` imports would have failed before, that's a better description.